src/librustc_typeck/check/method/suggest.rs

```diff
@@ -518,7 +518,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
 
-                if let Some(lev_candidate) = lev_candidate {
+                fn is_str_ref(ty: Ty<'_>) -> bool {
```

Comment:
Seems like `ty.is_str()` is missing and should be added and then you can use `ty.peel_refs().is_str()` to encode this check.

Reply:
Sure, that's a good one, I didn't realize there's a `peel_refs()` ready.